src/cpprealm/persisted.hpp

```diff
@@ -275,8 +302,251 @@ struct persisted_container_base : public persisted_base<T> {
+void persisted_container_base<T>::set(size_type pos, const typename T::value_type& a)
+requires (type_info::PrimitivePersistable<typename T::value_type>) {
+    if (this->m_obj) {
+        auto as_core_type = static_cast<typename type_info::persisted_type<typename T::value_type>::type>(a);
+        auto lst = this->m_obj->template get_list<typename type_info::persisted_type<typename T::value_type>::type>(this->managed);
+        lst.set(pos, as_core_type);
+    } else {
+        this->unmanaged[pos] = a;
+    }
+}
+
+template <realm::type_info::ListPersistable T>
+void persisted_container_base<T>::set(size_type pos, typename T::value_type& a)
+requires (type_info::ObjectPersistable<typename T::value_type>) {
+    if (this->m_obj) {
+        auto lst = this->m_obj->template get_list<typename type_info::persisted_type<typename T::value_type>::type>(this->managed);
+        if (!a.m_obj) {
+            T::value_type::schema::add(a, this->m_obj->get_table()->get_link_target(this->managed), nullptr);
+        }
+        lst.set(pos, a.m_obj->get_key());
+    } else {
+        this->unmanaged[pos] = a;
+    }
+}
+
+template <realm::type_info::ListPersistable T>
+void persisted_container_base<T>::set(size_type pos, typename T::value_type&& a)
+requires (type_info::ObjectPersistable<typename T::value_type>) {
+    set(pos, a);
+}
+
+template <realm::type_info::ListPersistable T>
+size_t persisted_container_base<T>::find(const typename T::value_type& a) requires (type_info::PrimitivePersistable<typename T::value_type>) {
+    if (this->m_obj) {
+        auto lst = this->m_obj->template get_list<typename type_info::persisted_type<typename T::value_type>::type>(this->managed);
+        if (auto size = lst.size()) {
+            if constexpr (type_info::BinaryPersistable<typename T::value_type>) {
+                BinaryData core_type = BinaryData(reinterpret_cast<const char *>(a.data()), a.size());
+                return lst.find_first(core_type);
+            } else {
+                return lst.find_first(a);
+            }
+        }
+    } else {
+        auto it = std::find(this->unmanaged.begin(), this->unmanaged.end(), a);
+        if (it != this->unmanaged.end()) {
+          return std::distance(this->unmanaged.begin(), it);
+        } else {
+            return realm::npos;
+        }
+    }
+}
+
+template <realm::type_info::ListPersistable T>
+size_t persisted_container_base<T>::find(const typename T::value_type& a)
+requires (type_info::ObjectPersistable<typename T::value_type>) {
+    if (this->m_obj) {
+        if (!a.m_obj.has_value()) {
+            return realm::npos;
+        }
+        auto lst = this->m_obj->template get_list<typename type_info::persisted_type<typename T::value_type>::type>(this->managed);
+        if (auto size = lst.size()) {
+            return lst.find_first((*a.m_obj).get_key());
+        }
+    } else {
+        // unmanaged objects in vectors aren't equatable.
+        return realm::npos;
```

Comment:
So that's not completely the case. Even though unmanaged objects that aren't `std::move`'d will be copied into the vector, users can still take references to the stored objects and compare them.